### ISSUE TITLE
Example how to retrain model on user data?

### ISSUE THREAD

**Issue by Youjin1985** [lantunes/CrystaLLM#18]
Could you please provide an example or dicumentation how to re-train existing model on user cifs?

**Comment by lantunes** [comment]
Hi, thanks for your message. 

Can you provide more details about what you are trying to do? If you would like to train a new model from scratch on your own set of CIFs, the instructions for preparing a custom CIF dataset are [here](https://github.com/lantunes/CrystaLLM?tab=readme-ov-file#using-your-own-cif-files).

However, fine-tuning an existing model on custom CIF files was not explored during development, and is outside the scope of this project. If you're interested in experimenting, I recommend checking out nanoGPT, which provides an [example](https://github.com/karpathy/nanoGPT/blob/93a43d9a5c22450bbf06e78da2cb6eeef084b717/config/finetune_shakespeare.py) of fine-tuning a pretrained natural language model (e.g., on Shakespeare text). This approach might adaptable for CIFs.

**Comment by lantunes** [comment]
Just to clarify, it is possible to fine-tune an existing model with the code in this repo. In your config file, set:
```
init_from: 'resume'
```
Make sure the `out_dir` points to the directory containing the pre-trained model checkpoint. Fine-tuning is typically done with a lower learning rate than was used during pre-training.

But, again, fine-tuning wasn't explored during the development of this project, so we can't say how well it will work in practice.
